packages/installations/src/helpers/generate-fid.ts

```diff
@@ -17,17 +17,31 @@
 
 import { bufferToBase64UrlSafe } from './buffer-to-base64-url-safe';
 
-/** Generates a new FID using random values from Web Crypto API. */
+export const VALID_FID = /^[cdef][\w-]{21}$/;
+
+/**
+ * Generates a new FID using random values from Web Crypto API.
+ * Returns an empty string if FID generation fails for any reason.
+ */
 export function generateFid(): string {
-  // A valid FID has exactly 22 base64 characters, which is 132 bits, or 16.5
-  // bytes. our implementation generates a 17 byte array instead.
-  const fidByteArray = new Uint8Array(17);
-  crypto.getRandomValues(fidByteArray);
+  try {
+    // A valid FID has exactly 22 base64 characters, which is 132 bits, or 16.5
+    // bytes. our implementation generates a 17 byte array instead.
+    const fidByteArray = new Uint8Array(17);
+    const crypto =
+      self.crypto || ((self as unknown) as { msCrypto: Crypto }).msCrypto;
```

Comment:
out of curiosity:
is this an attempt to fix the clients that currently create "AA=="?

Reply:
No, it's an unrelated fix for IE 11.

https://developer.mozilla.org/en-US/docs/Web/API/Window/crypto#Syntax